### PR TITLE
Implement NequIP model and MLIP wrapper

### PR DESCRIPTION
**Pull request by Satvik24511**
## Description

This PR implements the NequIP (E(3)-equivariant Graph Neural Network) model and a generic Machine Learning Interatomic Potential (MLIP) wrapper for DeepChem.

**Key Additions:**
1.  **`NequIP` Model (`deepchem/models/torch_models/nequip.py`)**:
    *   Implements the SE(3)-equivariant backbone using `SE3GraphConv` and `Fiber` layers.
    *   Supports computing total potential energy and atomic forces (via autograd).
    *   Utilizes spherical harmonics and equivariant basis layers from `deepchem.utils.equivariance_utils`.
2.  **`MLIPModel` Wrapper (`deepchem/models/torch_models/mlip.py`)**:
    *   A generic wrapper for MLIP implementations (like NequIP) to integrate them with `TorchModel`.
    *   Handles complex input/output structures (lists of graphs, atomic numbers).
    *   Implements a composite loss function for Energy and Forces (`energy_weight * loss_e + force_weight * loss_f`).
    *   Includes a robust label sanitizer to handle various DeepChem dataset formats.
3.  **Utils (`deepchem/utils/ase_utils.py`)**:
    *   Added utilities for bridging DeepChem and ASE (Atomic Simulation Environment) for MD simulations.

**Changes:**
- Added `NequIP` class.
- Added `MLIPModel` class.
- Added `deepchem/utils/ase_utils.py`.
- Updated `deepchem/models/torch_models/__init__.py` to export new models.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentations (modification for documents)

## Checklist
- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/__init__.py [hunk 1]
```diff
@@ -52,6 +52,8 @@
 from deepchem.models.torch_models.ChemCeption import ChemCeption
 from deepchem.models.torch_models.fno import FNO, FNOModel
 from deepchem.models.torch_models.lnn import LNN, LNNModel
+from deepchem.models.torch_models.mlip import MLIPModel
```
Comment (JoseAntonioSiguenza) [inline deepchem/models/torch_models/__init__.py]: please move this imports to dgl-dependent models

File: deepchem/models/torch_models/mlip.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    """
+    Internal wrapper to adapt TorchModel's list input to backbone's signature.
+    TorchModel passes inputs as a single list argument. This wrapper unpacks it.
+    """
+
+    def __init__(self, module: nn.Module):
+        super(_MLIPWrapper, self).__init__()
+        self.module = module
+
+    def forward(self, inputs):
```
Comment (JoseAntonioSiguenza) [inline deepchem/models/torch_models/mlip.py]: please add type annotations.

File: deepchem/models/torch_models/mlip.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        # Unpack the list [graph, atomic_numbers]
+        g, z = inputs
+        return self.module(g, z)
+
+
+class MLIPModel(TorchModel):
```
Comment (JoseAntonioSiguenza) [inline deepchem/models/torch_models/mlip.py:31]: Please add usage examples and type annotations.

File: deepchem/models/torch_models/__init__.py [hunk 4]
```diff
@@ -52,6 +52,8 @@
 from deepchem.models.torch_models.ChemCeption import ChemCeption
```
Comment (JoseAntonioSiguenza) [inline deepchem/models/torch_models/__init__.py:52]: In general, please add unit tests and include these in the docs.

**Comment by JoseAntonioSiguenza** [comment]
Can you please address the comments above, I'd be happy to make another review then. I am also concerned about the usage of `ase` library.

**Comment by Satvik24511** [comment]
I understand the concern regarding the usage of ase library. It is only used in ase_utils.py which is basically a utility file i made for easier Molecular Dynamics demonstration. If that is not something which is required in Deepchem then I can completely remove the file. That will have no effect whatsoever on the core implementation.

File: deepchem/models/torch_models/__init__.py (hunk 4 above)
Comment (Satvik24511) [reply deepchem/models/torch_models/__init__.py:52]: Can you elaborate a bit on what you mean by adding them in the docs

File: deepchem/models/torch_models/__init__.py (hunk 1 above)
Comment (Satvik24511) [reply deepchem/models/torch_models/__init__.py]: Moved MLIP/NequIP logic behind explicit DGL dependency and ensured imports are DGL-gated.

File: deepchem/models/torch_models/mlip.py (hunk 2 above)
Comment (Satvik24511) [reply deepchem/models/torch_models/mlip.py]: Added type annotations to mlip and nequip

File: deepchem/models/torch_models/mlip.py (hunk 3 above)
Comment (Satvik24511) [reply deepchem/models/torch_models/mlip.py:31]: Added minimal usage example
